### PR TITLE
UI standardize + tasks state integrated into top update goal UI

### PR DESCRIPTION
**Pull request by AndrewHC36**
Credit to Jasmine for aiding in standardizing and theming individual text themes.

TL;DR: Standardized text themes in `main.dart` and pre-fabricated widgets in `lib/widgets/*.dart`. (Use them please, for the common good)

### Standardized Text Themes (please use them)
- Check `main.dart` for all the new text themes. Please use these to clean up all the clutter and standardize text themes. 
- If you really need a custom text themes, at least use `.copyWith()` method to "inherit" the standard text themes.
- Note: if you see any remaining unstandardized text themes, could y'all also clean those up to conform to one of the standard text themes.
### Standardized Pre-fab Widgets (please also use them)
- Check `lib/widgets` directory for all the current supported pre-fabricated widgets that are commonly used.
- Use them whenever you can
- Some like `rounded_progress_bar` are created, even when there's a standard widget for it, since the standard widget does not conform with the figma design and cannot be modified easily.
## Features & fixes
- **Text themes:** Text themes are standardized with the help from designer team, Jasmine. (The chart of those text themes are in #catalog)
- **State integration:** Whenever users create a new task, the `GoalTaskModel` will automatically increment to the number of tasks from user input in the tasks widget. This will get "notified" to both inside `UpdateGoalWidget` and `GoalWidget`.
- **File mini re-organization:** Separated the update goal portion inside the `goal.dart` into `update_goal.dart` due to the utter messiness when utilizing providers.
- There's probably other minor stuff that can be found in the files

### CODE REVIEW

**Comment by AndrewHC36** [comment]
https://www.loom.com/share/c9dbccd1489743d39a768f6691e1b5d7?sid=2a05580d-e5cf-4edf-8002-ccd196f2cd5e

**Review by CodingPenguin — APPROVED** [review]
looks good. solid work
